File: intermediate-server/factory.intermediate.iot/src/main/java/ua/nure/makieiev/factory/intermediate/iot/service/impl/FilterWorkLogServiceImpl.java
package ua.nure.makieiev.factory.intermediate.iot.service.impl;

import com.google.gson.Gson;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import ua.nure.makieiev.factory.intermediate.iot.model.FilterUnit;
import ua.nure.makieiev.factory.intermediate.iot.model.FilterWorkLogInfo;
import ua.nure.makieiev.factory.intermediate.iot.service.FilterWorkLogService;

import java.util.Objects;

@Service
public class FilterWorkLogServiceImpl implements FilterWorkLogService {

    @Override
    public boolean save(FilterWorkLogInfo filterWorkLogInfo, String token) {
        Gson gson = new Gson();
        String jsonInfo = gson.toJson(filterWorkLogInfo);
        RestTemplate rest = new RestTemplate();
        HttpHeaders headers = new HttpHeaders();
        headers.add("Content-Type", "application/json");
        headers.add("Accept", "*/*");
        headers.add("Authorization", "Bearer " + token);
        HttpEntity<String> requestEntity = new HttpEntity<>(jsonInfo, headers);
        ResponseEntity<String> responseEntity = rest.exchange("https://arkproject.herokuapp.com/filter/worklog/add", HttpMethod.POST, requestEntity, String.class);
        return Objects.nonNull(responseEntity.getBody());
    }

    @Override
    public FilterUnit findFilterUnitByUnitId(long unitId, String token) {
        Gson gson = new Gson();
        RestTemplate rest = new RestTemplate();
        HttpHeaders headers = new HttpHeaders();
        headers.add("Content-Type", "application/json");
        headers.add("Accept", "*/*");
        headers.add("Authorization", "Bearer " + token);
        HttpEntity<String> requestEntity = new HttpEntity<>("", headers);
        ResponseEntity<String> responseEntity = rest.exchange("https://arkproject.herokuapp.com/filter/unit/id/" + unitId, HttpMethod.GET, requestEntity, String.class);
        return gson.fromJson(requestEntity.getBody(), FilterUnit.class);
    }

}

File: backend/src/main/kotlin/ua/nure/makieiev/ark/service/impl/FilterWorkLogServiceImpl.kt
package ua.nure.makieiev.ark.service.impl

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import ua.nure.makieiev.ark.model.entity.FilterWorkLog
import ua.nure.makieiev.ark.repository.FilterWorkLogRepository
import ua.nure.makieiev.ark.service.FilterWorkLogService
import java.util.*

@Service
class FilterWorkLogServiceImpl @Autowired constructor(private val filterWorkLogRepository: FilterWorkLogRepository) : FilterWorkLogService {

    override fun save(filterWorkLog: FilterWorkLog): FilterWorkLog {
        return filterWorkLogRepository.save(filterWorkLog)
    }

    override fun findById(id: Long): Optional<FilterWorkLog> {
        return filterWorkLogRepository.findById(id)
    }

    override fun findAll(): List<FilterWorkLog> {
        return filterWorkLogRepository.findAll() as MutableList
    }

    override fun findAllByFilterUnitId(filterUnitId: Long): List<FilterWorkLog> {
        return filterWorkLogRepository.findAllByFilterUnitId(filterUnitId)
    }

}